h/static/scripts/test/widget-controller-test.coffee

```diff
@@ -1,16 +1,20 @@
 {module, inject} = angular.mock
 
+events = require('../events')
+
 describe 'WidgetController', ->
   $scope = null
   fakeAnnotationMapper = null
   fakeAnnotationUI = null
   fakeAuth = null
   fakeCrossFrame = null
+  fakeDrafts = null
   fakeStore = null
   fakeStreamer = null
   fakeStreamFilter = null
   fakeThreading = null
   fakeGroups = null
+  lastSearchResult = null
```

Comment:
I'll have a look and see if I can refactor along these lines. I haven't felt the pain you mentioned in the client-side tests myself yet but I can see the hazard.